book/src/input/watching_input_events.md

```diff
@@ -0,0 +1,46 @@
+# Watching Input Events From Systems
+
+You can also directly listen/post directly on input [event channels](../concepts/event-channel.md). This can be useful if you need to listen to an key/action being pressed as it happens.
+
+```rust,edition2018,no_run,noplaypen
+# extern crate amethyst;
+# use amethyst::{
+#     ecs::{System, SystemData, Read},
+#     prelude::*,
+# };
+use amethyst::{
+    input::{InputEvent, StringBindings},
+    shrev::{EventChannel, ReaderId},
+};
+
+pub struct CustomInputSystem {
+    input_event_rid: ReaderId<InputEvent<StringBindings>>,
+}
+
+impl CustomInputSystem {
+    pub fn new(world: &mut World) -> Self {
+        <Self as System<'_>>::SystemData::setup(world);
+        let input_event_rid = world.fetch_mut::<EventChannel<InputEvent<StringBindings>>>().register_reader();
+        Self { input_event_rid }
+    }
+}
+
+impl<'a> System<'a> for CustomInputSystem {
+    type SystemData = Read<'a, EventChannel<InputEvent<StringBindings>>>;
```

Comment:
I suggest adding a newline between `type SystemData...` and `fn run(...` for readability.
```suggestion
    type SystemData = Read<'a, EventChannel<InputEvent<StringBindings>>>;
    
```